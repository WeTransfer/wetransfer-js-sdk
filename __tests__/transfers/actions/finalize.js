const routes = require('../../../src/config/routes');
const finalizeAction = require('../../../src/transfers/actions/finalize');
const WTError = require('../../../src/error');

describe('Finalize transfer action', () => {
  let finalize = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue({
      id: 'random-hash',
      message: 'Little kittens',
      state: 'uploading',
      url: 'https://we.tl/t-random-hash',
      files: [
        {
          id: 'random-hash',
          name: 'kittie.gif',
          size: 1024,
          type: 'file',
        },
      ],
      expires_at: '2018-01-01T00:00:00Z',
    });

    finalize = finalizeAction({
      routes,
      request: mocks.request,
    });
  });

  it('should create a finalize transfer request', async () => {
    const transfer = await finalize({ id: 'transfer-id' });
    expect(transfer).toMatchSnapshot();
  });

  it('should throw an exception if the request fails', async () => {
    mocks.request.send.mockImplementation(() =>
      Promise.reject(new Error('Network error.'))
    );

    try {
      await finalize();
    } catch (error) {
      expect(error).toBeInstanceOf(WTError);
    }
  });
});
