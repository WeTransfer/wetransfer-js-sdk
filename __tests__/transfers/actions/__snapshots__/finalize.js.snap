// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finalize transfer action should create a finalize transfer request 1`] = `
RemoteTransfer {
  "files": Array [
    Object {
      "id": "random-hash",
      "name": "kittie.gif",
      "size": 1024,
      "type": "file",
    },
  ],
  "id": "random-hash",
  "message": "Little kittens",
  "state": "uploading",
  "url": "https://we.tl/t-random-hash",
}
`;
